src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -298,7 +318,117 @@
                 }
 
                 return this;
+            },
+
+            createTabs: function() {
+                var tabGroups = document.querySelectorAll('.sf-tabs');
+
+                /* create the tab navigation for each group of tabs */
+                for (var i = 0; i < tabGroups.length; i++) {
+                    var tabs = tabGroups[i].querySelectorAll('.tab');
+                    var tabNavigation = document.createElement('ul');
+                    tabNavigation.setAttribute('class', 'tab-navigation');
+
+                    for (var j = 0; j < tabs.length; j++) {
+                        var tabId = 'tab-' + i + '-' + j;
+                        var tabTitle = tabs[j].querySelector('.tab-title').innerHTML;
+
+                        var tabNavigationItem = document.createElement('li');
+                        tabNavigationItem.setAttribute('data-tab-id', tabId);
+                        tabNavigationItem.className = (j == 0) ? 'active' : '';
+                        tabNavigationItem.innerHTML = tabTitle;
+                        tabNavigation.appendChild(tabNavigationItem);
+
+                        var tabContent = tabs[j].querySelector('.tab-content');
+                        tabContent.parentElement.setAttribute('id', tabId);
+                    }
+
+                    tabGroups[i].insertBefore(tabNavigation, tabGroups[i].firstChild);
+                }
+
+                /* display the active tab and add the 'click' event listeners */
+                for (i = 0; i < tabGroups.length; i++) {
+                    tabNavigation = tabGroups[i].querySelectorAll('.tab-navigation li');
+
+                    for (j = 0; j < tabNavigation.length; j++) {
+                        tabId = tabNavigation[j].getAttribute('data-tab-id');
+                        document.getElementById(tabId).querySelector('.tab-title').style.display = 'none';
+
+                        if (Sfjs.hasClass(tabNavigation[j], 'active')) {
+                            document.getElementById(tabId).style.display = 'block';
+                        } else {
+                            document.getElementById(tabId).style.display = 'none';
+                        }
+
+                        tabNavigation[j].addEventListener('click', function(e) {
+                            var activeTab = e.target || e.srcElement;
+
+                            /* needed because when the tab contains HTML contents, user can click */
+                            /* on any of those elements instead of their parent '<li>' element */
+                            while (activeTab.tagName.toLowerCase() !== 'li') {
+                                activeTab = activeTab.parentNode;
+                            }
+
+                            for (var k = 0; k < tabNavigation.length; k++) {
```

Comment:
I think the new code is less readable :(  By the way, a lot of comments are related to JavaScript performance. But please, let's focus a bit. This is not a full-screen 3-D game rendered with JavaScript + canvas that must run at 60 fps. The JavaScript code involved in each profiler page is negligible.